yt/funcs.py

```diff
@@ -1231,3 +1231,29 @@ def validate_object(obj, data_type):
         raise TypeError("Expected an object of '%s' type, received '%s'"
                         % (str(data_type).split("'")[1],
                         str(type(obj)).split("'")[1]))
+
+def validate_axis(axis):
+    err_msg = ("Expected axis of int or char type ( can be "
+               "[0, 1, 2, '0', '1', '2', 'x', 'y', 'z']), received '%s'.")
```

Comment:
This won't work for non-cartesian data. You probably want to populate this list from e.g. `ds.coordinates.axis_.id.keys()`.

Reply:
Thanks, fixed.